src/editor/components/modals/ReportModal/ReportModal.component.jsx

```diff
@@ -0,0 +1,442 @@
+import { useState, useEffect } from 'react';
+import Modal from '../Modal.jsx';
+import useStore from '@/store.js';
+import styles from './ReportModal.module.scss';
+import { Button, TextArea, Input } from '@/editor/components/components';
+import { useAuthContext } from '@/editor/contexts';
+import { GeospatialIcon } from '@/editor/icons';
+
+export const ReportModal = () => {
+  const setModal = useStore((state) => state.setModal);
+  const isOpen = useStore((state) => state.modal === 'report');
+  // Auth context available if needed in the future
+  useAuthContext();
+
+  const [formData, setFormData] = useState({
+    description: '',
+    location: '',
+    currentCondition: '',
+    problemStatement: '',
+    proposedSolutions: ''
+  });
+
+  const [errors, setErrors] = useState({
+    description: '',
+    problemStatement: '',
+    geoLocation: ''
+  });
+
+  const [touched, setTouched] = useState({
+    description: false,
+    problemStatement: false,
+    geoLocation: false
+  });
```

Comment:
You're starting to write your own form library at this point.
Did you look at using https://react-hook-form.com for example that is quite popular? react-hook-form is the library used by shadcn https://ui.shadcn.com/docs/components/form with the zod validation library https://zod.dev

Reply:
Thanks for this feedback. I can fix this later, it's good enough to keep as-is for now and then can replace in the future without affecting core flow or data